### DISCUSSION TITLE
Change existing domain

### DISCUSSION THREAD

**Discussion by thiagomendonca-eu** [chatwoot/chatwoot#8453] [Self hosted]
### Describe the bug

I performed the Self-Hosted installation from the AWS Marketplace and ran the command "sudo cwctl --webserver" to define a domain "chatwoot.domain.com" for my application and everything went correctly. However, I would like to change the domain to "chat.domain.com" and an error is occurring when executing "sudo cwctl --webserver". Would you like to know how to proceed? Thank you very much in advance!

---

    Do you wish to proceed? (yes or no): yes
    Reading package lists... Done
    Building dependency tree       
    Reading state information... Done
    nginx is already the newest version (1.18.0-0ubuntu1.4).
    certbot is already the newest version (0.40.0-1ubuntu0.1).
    nginx-full is already the newest version (1.18.0-0ubuntu1.4).
    python3-certbot-nginx is already the newest version (0.40.0-0ubuntu0.1).
    0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed
    100   769  100   769    0     0   2546      0 --:--:-- --:--:-- --:--:--  2546
    --2023-11-14 22:09:18--  https://raw.githubusercontent.com/chatwoot/chatwoot/develop/deployment/nginx_chatwoot.conf
    Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...
    Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.
    HTTP request sent, awaiting response... 200 OK
    Length: 1963 (1.9K) [text/plain]
    Saving to: ‘nginx_chatwoot.conf.2’
    
    nginx_chatwoot.conf.2                100%[======================================================================>]   1.92K  --.-KB/s    in 0s      
    
    2023-11-14 22:09:18 (21.7 MB/s) - ‘nginx_chatwoot.conf.2’ saved [1963/1963]
    
    Saving debug log to /var/log/letsencrypt/letsencrypt.log
    Error while running nginx -c /etc/nginx/nginx.conf -t.
    
    nginx: [emerg] cannot load certificate "/etc/letsencrypt/live/chatwoot.domain.com/fullchain.pem": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen('/etc/letsencrypt/live/chatwoot.domain.com/fullchain.pem','r') error:2006D080:BIO routines:BIO_new_file:no such file)
    nginx: configuration file /etc/nginx/nginx.conf test failed
    
    Could not choose appropriate plugin: The nginx plugin is not working; there may be problems with your existing configuration.
    The error was: MisconfigurationError('Error while running nginx -c /etc/nginx/nginx.conf -t.\n\nnginx: [emerg] cannot load certificate "/etc/letsencrypt/live/chatwoot.domain.com/fullchain.pem": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen(\'/etc/letsencrypt/live/chatwoot.domain.com/fullchain.pem\',\'r\') error:2006D080:BIO routines:BIO_new_file:no such file)\nnginx: configuration file /etc/nginx/nginx.conf test failed\n')
    The nginx plugin is not working; there may be problems with your existing configuration.
    The error was: MisconfigurationError('Error while running nginx -c /etc/nginx/nginx.conf -t.\n\nnginx: [emerg] cannot load certificate "/etc/letsencrypt/live/chatwoot.domain.com/fullchain.pem": BIO_new_file() failed (SSL: error:02001002:system library:fopen:No such file or directory:fopen(\'/etc/letsencrypt/live/chatwoot.domain.com/fullchain.pem\',\'r\') error:2006D080:BIO routines:BIO_new_file:no such file)\nnginx: configuration file /etc/nginx/nginx.conf test failed\n')
    
    Some error has occured. Check '/var/log/chatwoot-setup.log' for details.

### To Reproduce

1. Self-hosted
2. EC2 t3.medium instance from AWS Marketplace (https://www.chatwoot.com/docs/self-hosted/deployment/aws-marketplace).
3. Set a home domain with the command “sudo cwctl --webserver”
4. Try changing the domain with the same command.

### Environment

Other [please specify in the description]

### Cloud Provider

AWS

### Platform

Browser

**Comment by kalebwd** [comment]
I reported the same months ago and it doesn't seem there is much desire to investigate/fix this issue.

**Comment by jordan26** [comment]
I too want to do this in order to clone into a new server and change the domain for testing purposes.

**Comment by pranavrajs** [comment]
@thiagodebugs We don't have an option to update these at the moment. @vishnu-narayanan This might be a good one to add to the cwctl command.

**Comment by outrank-james** [comment]
Same issue here after a slight name change

**Comment by outrank-james** [comment]
Any news on an update / fix for this - been trying to get this server working again but fail every time I try. Only way I can see it working again is to rebuild the virtual env
